pandas/tests/frame/test_sorting.py

```diff
@@ -500,6 +526,71 @@ def test_sort_index_categorical_index(self):
         expected = df.iloc[[2, 3, 0, 1, 5, 4]]
         tm.assert_frame_equal(result, expected)
 
+    def test_sort_index_key(self):
```

Comment:
can you make a new test class to group the new tests

Reply:
done.